recipes/icu/build.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+cd source
+chmod +x configure install-sh
+
+./configure --prefix="$PREFIX" \
+    --disable-samples \
+    --disable-extras \
+    --disable-layout \
+    --enable-static
```

Comment:
> 1. Do we agree to add `--disable-tests` here? I think @jakirkham said tests only add some test data and don't have to do with actual tests (with `make check` is already doing).

Correct. `make check` is working great. This installed some weighty test data that I don't think we care about. It might have slipped into the last PR I did against @msarahan's branch. If it did, that was totally accidental and should be removed. :+1:

> Do we really need the static library? I'm -1 for it because it's not needed for Boost and Qt5, and it makes the package bigger.

IMHO we should always distribute static and dynamic libraries. I have known people that prefer to only use static libraries. For instance, the ATLAS developers [prefer static libraries]( http://math-atlas.sourceforge.net/atlas_install/node22.html ) so make this the default, but provide an option that one must enable to build the dynamic libraries. If someone would rather take the needed symbols from ICU to make their package distribution simpler, I am ok with that.